packages/statemanager/src/stateManager.ts

```diff
@@ -153,15 +269,25 @@ export class DefaultStateManager extends BaseStateManager implements StateManage
    */
   async getContractCode(address: Address): Promise<Buffer> {
     const account = await this.getAccount(address)
+    if (!account) {
+      return Buffer.alloc(0)
+    }
     if (!account.isContract()) {
       return Buffer.alloc(0)
     }
     const key = this._prefixCodeHashes
       ? Buffer.concat([CODEHASH_PREFIX, account.codeHash])
       : account.codeHash
-    // @ts-expect-error
-    const code = await this._trie._db.get(key)
-    return code ?? Buffer.alloc(0)
+
+    const keyHex = key.toString('hex')
+    if (keyHex in this._codeCache) {
+      return this._codeCache[keyHex]
+    } else {
+      // @ts-expect-error
+      const code = (await this._trie._db.get(key)) ?? Buffer.alloc(0)
```

Comment:
Just a question about best practice on Typescript errors like this, should we use the above `@ts-expect-error` which makes it not obvious to me what the actual error is (since VS Code respects this and doesn't give you the red squiggles), or should we do `(await (this._trie as any)._db...`?  I'm fine with this approach but maybe you could add a comment explaining what the expected error is (i.e. `db` is a protected member of the `trie` class)

Reply:
Just seeing, thanks for updating. 

Update: ah, and just also seeing that we have a public access method for the database in the Trie library (`database()`).

Will just update with this, so can prevent the whole thing.